src/nl_write.jl

```diff
@@ -187,7 +187,7 @@ function write_nl_j_blocks(f, m::AmplNLMathProgModel)
 
             # Assemble tuples of (.nl index, constraint value)
             output = collect(zip(
-                map(j->m.v_index_map[j], keys(m.lin_constrs[i])),
+                Set((m.v_index_map[j] for j in keys(m.lin_constrs[i]))), # TODO: is Set necessary here?
```

Comment:
`Set` was suggested to me by the deprecation warning. Is it actually necessary?

Reply:
Maybe just to avoid duplicates? I'm not sure

Reply:
Turned out following the depwarn's advice was definitely a bad idea, as it caused the minlp test failure.